platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/SolrClientAdapter.java

```diff
@@ -387,7 +389,7 @@ public boolean isAvailable(long timeout, TimeUnit unit) throws InterruptedExcept
         }
         final long timeRemaining = end - now;
 
-        if (timeRemaining <= 0L) { // we timed out
+        if (timeRemaining <= interval) { // we timed out
```

Comment:
:question: Won't this cause the retries to quick early?  Seems like if time remaining is 59 seconds and the interval is 60 then I would want to wait the remaining time and try one last time instead of giving up 59 seconds early.